scripts/dictcli.py

```diff
@@ -0,0 +1,281 @@
+    lang_map = valid_languages()
+    api_list = language_list_from_api()
+    code2file = {}
+    for code, filename in api_list:
+        if latest_yet(code2file, code, filename):
+            code2file[code] = filename
+    return [
+        Language(code, name, code2file[code])
+        for code, name in lang_map.items()
+        if code in code2file
+    ]
+
+
+def download_dictionary(url, dest):
+    """Download a decoded dictionary file."""
+    response = urllib.request.urlopen(url)
+    decoded = base64.decodebytes(response.read())
+    with open(dest, 'bw') as dict_file:
+        dict_file.write(decoded)
+
+
+def filter_languages(languages, selected):
+    """Filter a list of languages based on an inclusion list.
+
+    Args:
+        languages: a list of languages to filter
+        selected: a list of keys to select
+    """
+    filtered_languages = []
+    for language in languages:
+        if language.code in selected:
+            filtered_languages.append(language)
+            selected.remove(language.code)
+    if selected:
+        raise InvalidLanguageError(selected)
+    return filtered_languages
+
+
+def install_lang(lang):
+    """Install a single lang given by the argument."""
+    lang_url = API_URL + lang.remote_path + '?format=TEXT'
+    if not os.path.isdir(spell.dictionary_dir()):
+        msg = '{} does not exist, creating the directory'
+        print(msg.format(spell.dictionary_dir()))
+        os.makedirs(spell.dictionary_dir())
+    print('Downloading {}'.format(lang_url))
+    dest = os.path.join(spell.dictionary_dir(), lang.remote_path)
+    download_dictionary(lang_url, dest)
+    print('Done.')
+
+
+def install(languages):
+    """Install languages."""
+    for lang in languages:
+        try:
+            print('Installing {}: {}'.format(lang.code, lang.name))
+            install_lang(lang)
+        except PermissionError as e:
+            print(e)
+            sys.exit(1)
+
+
+def update(languages):
+    installed = [lang for lang in languages if lang.local_version is not None]
+    for lang in installed:
+        if lang.local_version < lang.remote_version:
+            print('Upgrading {} from {} to {}'.format(
+                lang.code,
+                version_str(lang.local_version),
+                version_str(lang.remote_version)))
+            install_lang(lang)
+
+
+def remove_old(languages):
+    installed = [lang for lang in languages if lang.local_version is not None]
+    for lang in installed:
+        local_files = spell.local_files(lang.code)
+        for old_file in local_files[1:]:
+            os.remove(os.path.join(spell.dictionary_dir(), old_file))
+
+
+def main():
+    if configdata.DATA is None:
+        configdata.init()
+    parser = get_argparser()
+    argv = sys.argv[1:]
+    args = parser.parse_args(argv)
+    languages = available_languages()
+    if args.cmd is None:
+        parser.print_usage()
+        exit(1)
```

Comment:
Please use `sys.exit`, the builtin `exit` is intended for the Python interactive REPL/shell.